src/app/base/side-panel-context-new.tsx

```diff
@@ -17,20 +17,55 @@ interface SidePanelState<TProps = Record<string, unknown>> {
 }
 
 interface SidePanelActions {
-  open: <TProps extends Record<string, unknown> = Record<string, unknown>>(
-    component: ComponentType<TProps>,
-    title: string,
-    props?: TProps,
-    size?: SidePanelSize
-  ) => void;
-  close: () => void;
-  setSize: (size: SidePanelSize) => void;
+  openSidePanel: <
+    TProps extends Record<string, unknown> = Record<string, unknown>,
+  >(params: {
+    component: ComponentType<TProps>;
+    title: string;
+    props?: TProps;
+    size?: SidePanelSize;
+  }) => void;
+  closeSidePanel: () => void;
+  setSidePanelSize: (size: SidePanelSize) => void;
 }
 
 type SidePanelContextValue = SidePanelActions & SidePanelState;
 
 const SidePanelContext = createContext<SidePanelContextValue | null>(null);
 
+/**
+ * Hook for managing side panel state and actions.
+ *
+ * Provides methods to open/close side panels with React components and manages
+ * panel size and visibility state.
+ *
+ * @returns Object containing:
+ *   - `isOpen`: Boolean indicating if the side panel is currently open
+ *   - `title`: Current panel title string
+ *   - `size`: Current panel size ('narrow' | 'regular' | 'wide' | 'large')
+ *   - `component`: Currently rendered component
+ *   - `props`: Props passed to the current component
+ *   - `open(component, title, props?, size?)`: Opens panel with given component
+ *   - `close()`: Closes the panel and resets state
+ *   - `setSize(size)`: Changes the panel width
```

Comment:
*Love* this, just make sure the method names match the actual returns

```suggestion
 * @returns Object containing:
 *   - `isOpen`: Boolean indicating if the side panel is currently open
 *   - `title`: Current panel title string
 *   - `size`: Current panel size ('narrow' | 'regular' | 'wide' | 'large')
 *   - `component`: Currently rendered component
 *   - `props`: Props passed to the current component
 *   - `openSidePanel({ component, title, props?, size? })`: Opens panel with given component
 *   - `closeSidePanel()`: Closes the panel and resets state
 *   - `setSidePanelSize(size)`: Changes the panel width
```